File: 2022/03/2203.py
import os
import string

ano, dia, input = "2022", "03", "input"
prioridade = string.ascii_lowercase + string.ascii_uppercase

os.chdir(f"{ano}/{dia}")
lines = open(f"{input}.txt").read().splitlines()

n = 0
for line in lines:
    length = int(len(line) / 2)
    items1 = line[0:length]
    items2 = line[length:]
    comum = ""
    for i1 in items1:
        for i2 in items2:
            if i1 == i2:
                comum = i1
    n += prioridade.index(comum[0]) + 1

print("PARTE 1:", n)

n = 0
for group in range(0, int(len(lines) / 3)):
    items1 = lines[0 + 3 * group]
    items2 = lines[1 + 3 * group]
    items3 = lines[2 + 3 * group]
    comum = ""
    for i1 in items1:
        for i2 in items2:
            for i3 in items3:
                if i1 == i2 == i3:
                    comum = i1
    n += prioridade.index(comum[0]) + 1

print("PARTE 2:", n)

File: 2022/03/2203.go
package main

import (
	"bufio"
	"fmt"
	"log"
	"os"
	"strings"
)

const year, day, input string = "2022", "03", "input"
const prioridade string = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"

func readFile() []string {
	file, err := os.Open(fmt.Sprintf("%v/%v/%v.txt", year, day, input))
	if err != nil {
		log.Fatal("erro na leitura do arquivo:", err)
	}

	scanner := bufio.NewScanner(file)

	lines := []string{}
	for scanner.Scan() {
		newLine := scanner.Text()
		lines = append(lines, newLine)
	}

	return lines
}

func main() {
	lines := readFile()
	var soma int
	for _, line := range lines {
		length := len(line)
		items1 := line[:length/2]
		items2 := line[length/2:]
		var comum string
		for _, i1 := range items1 {
			for _, i2 := range items2 {
				if i1 == i2 {
					comum = string(i1)
					goto outPart1
				}
			}
		}
	outPart1:
		prioridade := strings.Index(prioridade, comum)
		if prioridade >= 0 {
			soma += prioridade + 1
		}
	}
	fmt.Println("PARTE 1:", soma)

	soma = 0
	for i := 0; i < len(lines); i += 3 {
		items1 := lines[i]
		items2 := lines[i+1]
		items3 := lines[i+2]
		var comum string
		for _, i1 := range items1 {
			for _, i2 := range items2 {
				for _, i3 := range items3 {
					if (i1 == i2) && (i2 == i3) {
						comum = string(i1)
						goto outPart2
					}
				}
			}
		}
	outPart2:
		prioridade := strings.Index(prioridade, comum)
		if prioridade >= 0 {
			soma += prioridade + 1
		}
	}
	fmt.Println("PARTE 2:", soma)

}
